### PR TITLE
use repo-root relative path for unit test files

### PR DESCRIPTION
**Pull request by chliangGoogle**

### CODE REVIEW

File: FirebaseMessaging.podspec [hunk 1]
```diff
@@ -53,8 +53,8 @@ device, and it is completely free.
     unit_tests.source_files = 'Example/Messaging/Tests/*.[mh]'
     unit_tests.requires_app_host = true
     unit_tests.pod_target_xcconfig = {
-      # Unit tests do library imports using Firebase/Messaging relative paths.
-      'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}"/Firebase/Messaging',
+      # Unit tests do library imports using repo-root relative paths.
+      'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}"',
```
Comment (paulb777) [inline FirebaseMessaging.podspec]: Should be on the library build rule, not the testspec - like https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseABTesting.podspec#L40

Comment (chliangGoogle) [reply FirebaseMessaging.podspec]: That was my first attempt, but didn't work either. I moved it to the library build now.

File: FirebaseMessaging.podspec [hunk 2]
```diff
@@ -36,7 +36,9 @@ device, and it is completely free.
     'GCC_C_LANGUAGE_STANDARD' => 'c99',
     'GCC_PREPROCESSOR_DEFINITIONS' =>
       'GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1 ' +
-      'FIRMessaging_LIB_VERSION=' + String(s.version)
+      'FIRMessaging_LIB_VERSION=' + String(s.version),
+       # Unit tests do library imports using repo-root relative paths.
+      'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}"',
```
Comment (paulb777) [inline FirebaseMessaging.podspec]: `HEADER_SEARCH_PATHS` is not a GCC_PREPROCESSOR_DEFINITIONS. It should be a top-level setting.

Comment (chliangGoogle) [reply FirebaseMessaging.podspec]: Oops! Nice catch!

File: FirebaseMessaging.podspec (hunk 1 above)
Comment (chliangGoogle) [reply FirebaseMessaging.podspec]: Think I got it fixed now after I tuned the header search path in xcode a bit too.

File: Example/Firebase.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -6853,7 +6853,7 @@
 				DEBUG_INFORMATION_FORMAT = dwarf;
 				DEVELOPMENT_TEAM = 4ANB9W7R3P;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
-				INFOPLIST_FILE = $SRCROOT/DynamicLinks/FDLBuilderTestAppObjC/Info.plist;
+				INFOPLIST_FILE = "$SRCROOT/DynamicLinks/FDLBuilderTestAppObjC/Info.plist";
```
Comment (chliangGoogle) [inline Example/Firebase.xcodeproj/project.pbxproj]: this seems always automatically added when I do pod update, and pod deintegrate

File: Example/Messaging/Tests/FIRMessagingClientTest.m [hunk 4]
```diff
@@ -16,22 +16,24 @@
...
+#import "Firebase/Messaging/Protos/GtalkCore.pbobjc.h"
+
+#import "Firebase/Messaging/FIRMessagingClient.h"
+#import "Firebase/Messaging/FIRMessagingConnection.h"
+#import "Firebase/Messaging/FIRMessagingDataMessageManager.h"
+#import "Firebase/Messaging/FIRMessagingRmqManager.h"
+#import "Firebase/Messaging/FIRMessagingSecureSocket.h"
+#import "Firebase/Messaging/FIRMessagingUtilities.h"
+#import "Firebase/Messaging/NSError+FIRMessaging.h"
+#import "FIRMessagingFakeConnection.h"
```
Comment (paulb777) [inline Example/Messaging/Tests/FIRMessagingClientTest.m]: Changing all imports including ones in Test paths to repo relative could make the copybara rules simpler.

Comment (chliangGoogle) [reply Example/Messaging/Tests/FIRMessagingClientTest.m]: I also realize I need to update the path for library also.

Comment (chliangGoogle) [reply Example/Messaging/Tests/FIRMessagingClientTest.m]: Let me submit this one and do in another PR. This one also fixes some flakyness in tvOS tests.

**Pull request merged**